src/details/ArborX_DetailsUtils.hpp

```diff
@@ -526,9 +526,21 @@ void reallocWithoutInitializing(View &v,
                                 size_t n6 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
                                 size_t n7 = KOKKOS_IMPL_CTOR_DEFAULT_ARG)
 {
-  static_assert(View::is_managed, "Can only realloc managed views");
-  v = View(Kokkos::view_alloc(Kokkos::WithoutInitializing, v.label()), n0, n1,
-           n2, n3, n4, n5, n6, n7);
+  size_t ranks[8] = {n0, n1, n2, n3, n4, n5, n6, n7};
+  bool allocation_necessary = false;
+  for (unsigned int dim = 0; dim < v.rank_dynamic; ++dim)
```

Comment:
Does this change the behavior of the function compared to the original? Before, we did not have this check.

Reply:
That we didn't have this check before is the point of this pull request, isn't it? :wink:

The "original", i.e. `Kokkos::realloc`, doesn't have this check, yet.

This pull request shouldn't change the observable behavior of this function: We get a `Kokkos::View` that has the prescribed dimensions. Tools, on the other hand, can of course detect that we avoid an allocation and a deallocation.